File: api/Models/Medicine.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models;

public class Medicine
{
    public Guid Id { get; set; }

    [Required]
    public string FullName { get; set; } = string.Empty;

    public string Notes { get; set; } = string.Empty;

    public DateTime ExpiryDate { get; set; }

    public int Quantity { get; set; }

    public decimal Price { get; set; }

    [Required]
    public string Brand { get; set; } = string.Empty;
}

File: client/src/models/medicine.ts
export type Medicine = {
  id: string
  fullName: string
  notes: string
  expiryDate: string
  quantity: number
  price: number
  brand: string
}

export type MedicineFormState = {
  fullName: string
  notes: string
  expiryDate: string
  quantity: string
  price: string
  brand: string
}

export const emptyMedicineForm: MedicineFormState = {
  fullName: '',
  notes: '',
  expiryDate: '',
  quantity: '',
  price: '',
  brand: '',
}
